test_describe_with_dirty_suffix: fails on py2 on i686

```python
======================================================================
FAIL: test_describe_with_dirty_suffix (test.test_describe.DescribeDirtyWorkdirTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/pygit2-0.23.3/test/test_describe.py", line 106, in test_describe_with_dirty_suffix
    self.assertEqual('thetag-dirty', self.repo.describe(dirty_suffix='-dirty'))
AssertionError: u'thetag-dirty' != u'thetag\x08'
- thetag-dirty
+ thetag
------------
```

--- a/pygit2/repository.py
+++ b/pygit2/repository.py
@@ -716,8 +716,10 @@ class Repository(_Repository):
                 format_options.abbreviated_size = abbreviated_size
             if always_use_long_format is not None:
                 format_options.always_use_long_format = always_use_long_format
+            dirty_ptr = None
             if dirty_suffix:
-                format_options.dirty_suffix = ffi.new('char[]', to_bytes(dirty_suffix))
+                dirty_ptr = ffi.new('char[]', to_bytes(dirty_suffix))
+                format_options.dirty_suffix = dirty_ptr
 
             buf = ffi.new('git_buf *', (ffi.NULL, 0))
 

